x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -99,6 +100,47 @@ func parse(b []byte) ([]statsdMetric, error) {
 	return metrics, nil
 }
 
+func eventMapping(metricName string, metricValue interface{}, metricSetFields common.MapStr, mappings map[string]StatsdMapping) {
+	if len(mappings) == 0 {
+		metricSetFields[common.DeDot(metricName)] = metricValue
+		return
+	}
+
+	for _, mapping := range mappings {
+		// The metricname match the one with no labels in mappings
+		// Let's insert it dedotted and continue
+		if metricName == mapping.Metric {
+			metricSetFields[mapping.Value.Field] = metricValue
+			return
+		}
+
+		res := mapping.regex.FindStringSubmatch(metricName)
+
+		// Not all labels match
+		// Skip and continue to next mapping
+		if len(res) != (len(mapping.Labels) + 1) {
+			continue
```

Comment:
Will this drop metrics? Maybe we should log something at the debug level.

Reply:
This is the case where there's no match between the regex pattern and the input metric because the regex has more labels that the ones found in the metric. Now that we have validation for the config it should be impossible to happen, anyway I'd keep handling this case. I will add a debug log